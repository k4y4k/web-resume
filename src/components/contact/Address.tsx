import * as React from "react";
import tw, { css } from "twin.macro";
import { FiMapPin } from "@react-icons/all-files/fi/FiMapPin";

interface AddressTypes {
  city: string;
  countryCode: string;
  region: string;
  restrictDisplay?: boolean;
}

const AddressStyles = css`
  ${tw`whitespace-normal! flex flex-row flex-nowrap align-middle`}

  .icon {
    ${tw`flex-grow`}
    margin-right: 8px;
  }
`;

const Address = ({
  city,
  countryCode,
  region,
}: AddressTypes): JSX.Element | null => {
  // if there isn't a 123 Example Rd-type datum (non-restricted) or
  // a city (restricted), assume there's nothing else
  if (city === "") return null;

  // restricted mode enabled by default
  return (
    <li css={AddressStyles} data-testid="contactAddress" id="address">
      <FiMapPin />
      <span>
        {city}, {region + ","} {countryCode}
      </span>
    </li>
  );
};

export default Address;
