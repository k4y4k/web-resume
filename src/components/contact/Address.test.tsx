import * as React from "react";
import { render, screen, within } from "@testing-library/react";
import Address from "./Address";

describe("<Address />", () => {
  test("Handles no data", () => {
    render(<Address city="" countryCode="" region="" />);

    const address = screen.queryByTestId("contactAddress");

    expect(address).not.toBeInTheDocument();
  });

  describe("Is formatted correctly", () => {
    test("street name + number on one line", () => {
      render(
        <Address
          city="Example City"
          countryCode="EX"
          region="Example State"
          restrictDisplay={false}
        />
      );

      const address = screen.getByTestId("contactAddress");

      expect(address).toHaveTextContent("Example State");
      expect(address).toMatchSnapshot();
    });

    test("city, state on one line", () => {
      render(
        <Address
          city="Example City"
          countryCode="ZZ"
          restrictDisplay={false}
          region="Example State"
        />
      );

      const addressContainer = document.getElementById(
        "address"
      ) as HTMLElement;
      const address = within(addressContainer).getByText(
        "Example City, Example State",
        { exact: false }
      );

      expect(address).not.toBeFalsy();
      expect(address).toMatchSnapshot();
    });
  });

  describe("Is not about to dox me", () => {
    test("only displays City / State when RESTRICT_ADDRESS env var is passed in", () => {
      process.env.RESTRICT_ADDRESS = "true";

      render(
        <Address city="Example City" countryCode="ZZ" region="Example State" />
      );

      const address = screen.getByTestId("contactAddress");

      expect(address).not.toBeFalsy();
      expect(address).not.toHaveTextContent("Example Rd");
      expect(address).toHaveTextContent("Example City, Example State");
      expect(address).toMatchSnapshot();
    });
  });
});
