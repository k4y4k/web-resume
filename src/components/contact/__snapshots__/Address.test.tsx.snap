// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> Is formatted correctly city, state on one line 1`] = `
<span
  class="ml-[3px]"
>
  Example City
  , 
  Example State
</span>
`;

exports[`<Address /> Is formatted correctly street name + number on one line 1`] = `
<li
  class="!whitespace-normal flex flex-row flex-nowrap align-middle"
  data-testid="contactAddress"
  id="address"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
    />
    <circle
      cx="12"
      cy="10"
      r="3"
    />
  </svg>
  <span
    class="ml-[3px]"
  >
    Example City
    , 
    Example State
  </span>
</li>
`;

exports[`<Address /> Is not about to dox me only displays City / State when RESTRICT_ADDRESS env var is passed in 1`] = `
<li
  class="!whitespace-normal flex flex-row flex-nowrap align-middle"
  data-testid="contactAddress"
  id="address"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
    />
    <circle
      cx="12"
      cy="10"
      r="3"
    />
  </svg>
  <span
    class="ml-[3px]"
  >
    Example City
    , 
    Example State
  </span>
</li>
`;
